### DISCUSSION TITLE
Unable to publish package

### DISCUSSION THREAD

**Discussion by badgerowluke** [community/community#25395] [Actions]
<p>I’m fighting on getting a nuget package published into GPR.  Actions keeps flipping between unknown response and 401, any advice to help me clear this hurdle would be awesome:</p>
<pre><code class="lang-auto">
Run dotnet nuget push ../packages/*.nupkg --source "https://nuget.pkg.github.com/badgerowluke/index.json" -k ***
info : Pushing NOSQLORM.0.3.4-38-g320af3d.nupkg to 'https://nuget.pkg.github.com/badgerowluke'...
info :   PUT https://nuget.pkg.github.com/badgerowluke/
warn : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
info :   Unauthorized https://nuget.pkg.github.com/badgerowluke/ 436ms
error: Response status code does not indicate success: 401 (Unauthorized).
##[error]Process completed with exit code 1.
</code></pre>
<pre><code class="lang-auto">name: .NET Core

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:

    - uses: actions/checkout@v2
      with:
        fetch-depth: 0
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 2.1.806

    - run: sed 's/TOKEN/${{secrets.GITHUB_TOKEN}}/'  ./nuget.config

    - name: Install dependencies
      run: dotnet restore

    - name: Build
      run: dotnet build --configuration Release --no-restore

    - name: Test
      run: dotnet test --no-restore --verbosity normal /p:CollectCoverage=true  /p:CoverletOutputFormat=cobertura   
    - name: get_version

      run: |
        echo ::set-env name=VERSION_TAG::$(git describe --tags)

    - name: package
      run: dotnet pack -p:PackageId=NOSQLORM -p:PackageVersion=$VERSION_TAG --configuration Release --no-restore -o "../../packages"

    - name: publish
      run: |

        dotnet nuget push ../packages/*.nupkg --source "https://nuget.pkg.github.com/badgerowluke/index.json" -k ${{ secrets.GITHUB_TOKEN }}
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/badgerowluke">@badgerowluke</a>,<br>
I noticed the below step in your workflow,</p>
<pre><code class="lang-yaml">- run: sed 's/TOKEN/${{secrets.GITHUB_TOKEN}}/'  ./nuget.config
</code></pre>
<p>looks like, you directly replace the <strong>TOKEN</strong> with the <strong>GITHUB_TOKEN</strong> secret in the <strong>nuget.config</strong> file.</p>
<p>In most cases, directly using the <strong>GITHUB_TOKEN</strong> secret in the config or setting files is not allowed. You need to set the <strong>GITHUB_TOKEN</strong> secret as an environment variable, then pass the environment variable into the config/setting files. Other secrets set in the GitHub repository are similar.<br>
So, in your case, you can try like as below:</p>
<pre><code class="lang-yaml">- name: Pass token
  run: sed 's/TOKEN/%GITHUB_TOKEN%/'  ./nuget.config
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>The following is a simple example as reference:<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/.github/workflows/CI.yml" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/.github/workflows/CI.yml" target="_blank" rel="nofollow noopener">TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/.github/workflows/CI.yml</a></h4>
<pre><code class="lang-yml">name: Publish NuGet Package

on: push

env:
  MAIN_VER: 1

jobs:
  publish:
    name: Publish
    runs-on: windows-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Publish to GitHub Packages
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          cd TestNuGet
</code></pre>

  This file has been truncated. <a href="https://github.com/TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/.github/workflows/CI.yml" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/TestNuGet/nuget.config" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/TestNuGet/nuget.config" target="_blank" rel="nofollow noopener">TestGitHubPackagesRML/TestNuGet/blob/1bf4441f011b39ce4ddd659968db7acc7a05d73e/TestNuGet/nuget.config</a></h4>
<pre><code class="lang-config">&lt;?xml version="1.0" encoding="utf-8"?&gt;
&lt;configuration&gt;
    &lt;packageSources&gt;
        &lt;clear /&gt;
        &lt;add key="github" value="https://nuget.pkg.github.com/testgithubpackagesrml/index.json" /&gt;
    &lt;/packageSources&gt;
    &lt;packageSourceCredentials&gt;
        &lt;github&gt;
            &lt;add key="Username" value="Bright Ran" /&gt;
            &lt;add key="ClearTextPassword" value="%GITHUB_TOKEN%" /&gt;
        &lt;/github&gt;
    &lt;/packageSourceCredentials&gt;
&lt;/configuration&gt;
</code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
</p>

**Comment by badgerowluke** [comment]
<p>Thanks man! That helps a ton</p>
